packages/core/types/index.js

```diff
@@ -1008,14 +1008,19 @@ export type BundleResult = {|
 /**
  * @section resolver
  */
-export type ResolveResult = {|
-  +filePath?: FilePath,
-  +isExcluded?: boolean,
-  /** Corresponds to BaseAsset's <code>sideEffects</code>. */
-  +sideEffects?: boolean,
-  /** A resolver might want to resolve to a dummy, in this case <code>filePath</code> is rather "resolve from". */
-  +code?: string,
-|};
+export type ResolveResult =
+  | {|isExcluded: true|}
+  | {|
+      /** The result (an absolute path) */
+      +filePath: FilePath,
+      /** An unique string to be used instead of <code>path.relative(options.projectRoot, filePath)</code> when generating a hash, e.g. `${pkgName}@${version}/${subPath}` */
+      +cachePath?: string,
```

Comment:
Maybe we should rename this since it's not just used for the cache, but as part of the asset id? `uniqueKey` or something? 🤷